### PR TITLE
PLANET-7671 Add post type filter to News & Stories page

### PR DESCRIPTION
**Pull request by mleray**
### Description

See [PLANET-7671](https://jira.greenpeace.org/browse/PLANET-7671)

This is in addition to the previously added category filter

**Note:** for the mobile version we don't use the modal in the end.

### Testing

You can check out the new filter and "Apply" button on the [News & Stories page](https://www-dev.greenpeace.org/test-leda/news-stories/) from the leda test instance or on local. Make sure to test several filter combinations and also all screen sizes.

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [leda](https://www-dev.greenpeace.org/test-leda) | [admin](https://www-dev.greenpeace.org/test-leda/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-leda/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-leda) | [composer-local.json](https://github.com/greenpeace/planet4-test-leda/blob/main/composer-local.json)

:watch: 2024.12.04 15:05:59

File: templates/listing-page-filters.twig [hunk 1]
```diff
@@ -14,4 +14,17 @@
             </option>
         {% endfor %}
     </select>
+    <select class="form-select listing-page-select" id="post-type">
+        <option value="">{{__( 'All posts', 'planet4-master-theme' ) }}</option>
+        {% for key,post_type in post_types %}
+            {% set isSelected = current_post_type == post_type.slug %}
```
Comment (GP-Dan-Tovbein) [inline templates/listing-page-filters.twig:20]: What do you think to change to 
```suggestion
            {% set isSelected = current_post_type === post_type.slug %}
```

File: templates/listing-page-filters.twig [hunk 2]
```diff
@@ -14,4 +14,17 @@
...
+            <option
+                value="{{post_type.slug}}"
+                {{isSelected ? 'selected' : ''}}
```
Comment (GP-Dan-Tovbein) [inline templates/listing-page-filters.twig:23]: I think in this case could be better if we keep the attribute with the proper boolean.
```suggestion
    selected={{isSelected}}
```
What do you think?

**Review by GP-Dan-Tovbein — COMMENTED** [review]
Code looks good to me! I've left small comments which are not required to change.

I grabbed a video to show you the only "issue" I've found. 

https://github.com/user-attachments/assets/20da06e6-5fdf-4c77-84b9-31d189fdca6c

Do you think we can add also the layout as a param?

**Comment by mleray** [comment]
> Do you think we can add also the layout as a param?

@comzeradd what do you think about this? Should we try to find a way to also preserve the layout (grid or list) when filtering?

**Comment by comzeradd** [comment]
> @comzeradd what do you think about this? Should we try to find a way to also preserve the layout (grid or list) when filtering?

Yes, but if it requires refactoring some of the layout picker code let's do that in a follow up ticket.

**Comment by mleray** [comment]
@GP-Dan-Tovbein @comzeradd I've created a [ticket](https://jira.greenpeace.org/browse/PLANET-7706) for keeping the layout after reloading

**Review by GP-Dan-Tovbein — APPROVED** [review]
Code looks good to me 🥳

File: templates/listing-page-filters.twig (hunk 1 above)
Comment (mleray) [reply templates/listing-page-filters.twig:20]: We usually use `==` in twig templates, I see only in the google tag manager template we use `===` in one place 🤔

File: templates/listing-page-filters.twig (hunk 2 above)
Comment (mleray) [reply templates/listing-page-filters.twig:23]: This doesn't work, it results in this:

Screenshot 2024-12-17 at 15 20 56

**Pull request merged**